### PR TITLE
Switch to Drone CI

### PR DESCRIPTION
**Pull request by geek1011**
I've been getting annoyed with the long AppVeyor wait times for builds. Drone is also a lot more flexible for this, and supports parallel tasks with dependencies. The increased structure also helps with reading the logs.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  image: golang:1.12-stretch
+  commands:
+  - go build ./jiup/rules/reachability-test
+- name: test
+  image: debian:stretch
+  commands:
+  - apt update -qq
+  - apt install -qqy jq ca-certificates
+  - ./reachability-test --alternate-output
+# todo: decide on conditions to run this and updating the registry
```
Comment (geek1011) [inline .drone.yml]: ^

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  - git add just-install.json
+  - cat message.txt
+  - git commit -F message.txt || true
+  depends_on: [ update ]
+#-name: push
+# image: golang:1.12-stretch
+# commands:
+# - cd registry
+# - git push "https://$GITHUB_TOKEN@github.com/just-install/registry.git" master >/dev/null 2>/dev/null
+# depends_on: [ commit ]
```
Comment (geek1011) [inline .drone.yml]: I'll enable this once the PR is ready. Also, note to self about adding conditions to this.

**Comment by lvillani** [comment]
I would prefer to work with AppVeyor to resolve our pain points rather than increase the number of different CI solutions that we use. We just managed to consolidate everything under AppVeyor, I'm not sure I would want to undo that. I will ping AppVeyor about queue times this weekend.

**Comment by geek1011** [comment]
This also does reduce build times significantly, as the Drone runners are quite a bit more powerful. Also, I have had almost none of the timeouts compared to AppVeyor when running on Drone.

**Comment by lvillani** [comment]
Feel free to go for it then :)

**Comment by lvillani** [comment]
Hey! How is this going? Did you also happen to give GitHub Actions a shot?

**Comment by geek1011** [comment]
It's basically done (I just haven't got around to working on it recently). I just need to do the GH token and decide on when I'll show a failing status. I'll probably also rewrite the config in starlark, as it's more flexible for this.

Here's what I'm planning for now:

| What | Commit status | CI status | Push changes |
| --- | --- | --- | --- |
| Commit/rebuild | Same as CI | Fail if tests failed | Only if CI passes |
| Cron | Unchanged | Fail if tests failed | Only if CI passes |
| Force build | Unchanged | Pass unless push error | Always |

In the future, I might have the rules pass/fail/push individually.

**Comment by geek1011** [comment]
I'm going to close this for now, and include it as part of the refactor I've been working on intermittently.

**Pull request closed without merging**